packages/block-library/src/query-pagination-previous/index.php

```diff
@@ -44,13 +45,25 @@ function render_block_core_query_pagination_previous( $attributes, $content, $bl
 		add_filter( 'previous_posts_link_attributes', $filter_link_attributes );
 		$content = get_previous_posts_link( $label );
 		remove_filter( 'previous_posts_link_attributes', $filter_link_attributes );
-	} elseif ( 1 !== $page ) {
-		$content = sprintf(
-			'<a href="%1$s" %2$s>%3$s</a>',
-			esc_url( add_query_arg( $page_key, $page - 1 ) ),
-			$wrapper_attributes,
-			$label
-		);
+	} elseif ( 1 < $page ) {
+
+		$total_pages = $max_page;
+		if ( ! isset( $block->context['query']['inherit'] ) || ! $block->context['query']['inherit'] ) {
```

Comment:
The reason I suggested to move this block here is that we could avoid this `if` condition, since it's actually the `else` from above. 

I know realized that we have an `elseif ( 1 !== $page )`, so I guess we could change the `elseif` to `else`, and inside that `else` block handle the `if ( 1 !== $page )`.

This way we wouldn't also need the temp assignment `$total_pages = $max_page;`

I hope I didn't confuse you :)